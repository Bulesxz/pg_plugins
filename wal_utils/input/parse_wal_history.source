-- Install extension for tests
CREATE EXTENSION wal_utils;

CREATE TABLE history_data (data text);

-- Note that COPY truncates newlines.
-- Load history file that succeeds
COPY history_data FROM '@abs_srcdir@/data/example_success.history' WITH (FORMAT csv);
SELECT string_agg(data, E'\n') FROM history_data;
SELECT parse_wal_history(string_agg(data, E'\n')) FROM history_data;
DROP TABLE history_data;
