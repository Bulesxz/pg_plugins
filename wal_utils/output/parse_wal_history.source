-- Install extension for tests
CREATE EXTENSION wal_utils;
CREATE TABLE history_data (data text);
-- Note that COPY truncates newlines.
-- Load history file that succeeds
COPY history_data FROM '@abs_srcdir@/data/example_success.history' WITH (FORMAT csv);
SELECT string_agg(data, E'\n') FROM history_data;
                      string_agg                      
------------------------------------------------------
 1       0/09D4F390      no recovery target specified+
 2       0/117BEB70      no recovery target specified+
 3       0/187BEB38      no recovery target specified+
 4       0/188BEB38      no recovery target specified+
 5       0/189BEB38      no recovery target specified+
 7       0/249BEB38      no recovery target specified
(1 row)

SELECT parse_wal_history(string_agg(data, E'\n')) FROM history_data;
     parse_wal_history     
---------------------------
 (1,,0/9D4F390)
 (2,0/9D4F390,0/117BEB70)
 (3,0/117BEB70,0/187BEB38)
 (4,0/187BEB38,0/188BEB38)
 (5,0/188BEB38,0/189BEB38)
 (7,0/189BEB38,0/249BEB38)
(6 rows)

DROP TABLE history_data;
