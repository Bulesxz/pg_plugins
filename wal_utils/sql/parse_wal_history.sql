-- Install extension for tests
CREATE EXTENSION wal_utils;

CREATE TABLE history_data (data text);

-- Note that COPY truncates newlines.
-- Load history file that succeeds
INSERT INTO history_data VALUES (
'# Ignore this comment
1       0/09D4F390      no recovery target specified
2       0/117BEB70      no recovery target specified
3       0/187BEB38      no recovery target specified
# Other this comment and empty line

4       0/188BEB38      no recovery target specified
5       0/189BEB38      no recovery target specified
7       0/249BEB38      no recovery target specified
');
SELECT parse_wal_history(data) FROM history_data;
DROP TABLE history_data;
