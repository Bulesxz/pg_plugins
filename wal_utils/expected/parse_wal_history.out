-- Install extension for tests
CREATE EXTENSION wal_utils;
CREATE TABLE history_data (data text);
-- Note that COPY truncates newlines.
-- Load history file that succeeds
INSERT INTO history_data VALUES (
'# Ignore this comment
1       0/09D4F390      no recovery target specified
2       0/117BEB70      no recovery target specified
3       0/187BEB38      no recovery target specified
# Other this comment and empty line

4       0/188BEB38      no recovery target specified
5       0/189BEB38      no recovery target specified
7       0/249BEB38      no recovery target specified
');
SELECT parse_wal_history(data) FROM history_data;
     parse_wal_history     
---------------------------
 (1,,0/9D4F390)
 (2,0/9D4F390,0/117BEB70)
 (3,0/117BEB70,0/187BEB38)
 (4,0/187BEB38,0/188BEB38)
 (5,0/188BEB38,0/189BEB38)
 (7,0/189BEB38,0/249BEB38)
(6 rows)

DROP TABLE history_data;
